File: MTGSearch/src/androidTest/kotlin/com/dbottillo/mtgsearchfree/matchers/ViewMatchers.java
package com.dbottillo.mtgsearchfree.matchers;

import android.support.test.espresso.matcher.BoundedMatcher;
import android.view.View;

import com.dbottillo.mtgsearchfree.model.MTGCard;
import com.dbottillo.mtgsearchfree.model.MTGSet;

import org.hamcrest.Description;
import org.hamcrest.Matcher;

public final class ViewMatchers {

    private ViewMatchers(){

    }

    public static Matcher<Object> setWithName(final String name) {

        return new BoundedMatcher<Object, MTGSet>(MTGSet.class) {
            @Override
            public void describeTo(Description description) {
                description.appendText("with text: " + name);
            }

            @Override
            protected boolean matchesSafely(MTGSet item) {
                return name.equals(item.getName());
            }

        };
    }

    public static Matcher<Object> cardWithName(final String name) {

        return new BoundedMatcher<Object, MTGCard>(MTGCard.class) {
            @Override
            public void describeTo(Description description) {
                description.appendText("with text: " + name);
            }

            @Override
            protected boolean matchesSafely(MTGCard item) {
                return name.equals(item.getName());
            }

        };
    }

    public static Matcher<View> withDrawable(final int resourceId) {
        return new DrawableMatcher(resourceId);
    }

    public static Matcher<View> noDrawable() {
        return new DrawableMatcher(-1);
    }

}

File: app/src/androidTest/kotlin/com/dbottillo/mtgsearchfree/matchers/ViewMatchers.kt
package com.dbottillo.mtgsearchfree.matchers

import android.support.test.espresso.matcher.BoundedMatcher
import android.view.View

import com.dbottillo.mtgsearchfree.model.MTGCard
import com.dbottillo.mtgsearchfree.model.MTGSet

import org.hamcrest.Description
import org.hamcrest.Matcher

object ViewMatchers {

    fun setWithName(name: String): Matcher<Any> {

        return object : BoundedMatcher<Any, MTGSet>(MTGSet::class.java) {
            override fun describeTo(description: Description) {
                description.appendText("with text: $name")
            }

            override fun matchesSafely(item: MTGSet): Boolean {
                return name == item.name
            }
        }
    }

    fun cardWithName(name: String): Matcher<Any> {

        return object : BoundedMatcher<Any, MTGCard>(MTGCard::class.java) {
            override fun describeTo(description: Description) {
                description.appendText("with text: $name")
            }

            override fun matchesSafely(item: MTGCard): Boolean {
                return name == item.name
            }
        }
    }

    fun withDrawable(resourceId: Int): Matcher<View> {
        return DrawableMatcher(resourceId)
    }

    fun noDrawable(): Matcher<View> {
        return DrawableMatcher(-1)
    }
}
